examples/workflow.py

```diff
@@ -0,0 +1,230 @@
+"""Example for Max's Alginate data"""
+import redsky.event_streams as es
+from redsky.event_streams import dstar, star
+from streams.core import Stream
+import numpy as np
+from skbeam.core.accumulators.binned_statistic import BinnedStatistic1D
+from functools import partial
+# from xpdan.tools import better_mask_img
+from databroker.databroker import DataBroker as db
+from bluesky.callbacks.broker import LiveImage, LiveSliderImage
+from pprint import pprint
+from matplotlib.colors import LogNorm
+
+
+def subs(img1, img2):
+    return img1 - img2
+
+
+def add(img1, img2):
+    return img1 + img2
+
+
+def pull_array(img2):
+    return img2
+
+
+def generate_binner(geo, mask):
+    img_shape = mask.shape
+    r = geo.rArray(img_shape)
+    q = geo.qArray(img_shape) / 10
+    q_dq = geo.deltaQ(img_shape) / 10
+
+    pixel_size = [getattr(geo, a) for a in ['pixel1', 'pixel2']]
+    rres = np.hypot(*pixel_size)
+    rbins = np.arange(np.min(r) - rres / 2., np.max(r) + rres / 2., rres / 2.)
+    rbinned = BinnedStatistic1D(r.ravel(), statistic=np.max, bins=rbins, )
+
+    qbin_sizes = rbinned(q_dq.ravel())
+    qbin_sizes = np.nan_to_num(qbin_sizes)
+    qbin = np.cumsum(qbin_sizes)
+    return BinnedStatistic1D(q.flatten, bins=qbin, mask=mask.flatten())
+
+
+def z_score_image(img, binner):
+    xy = binner.xy
+    for i in np.unique(xy):
+        tv = (xy == i)
+        img[tv] -= np.mean(img[tv])
+        img[tv] /= np.std(img[tv])
+    return img
+
+
+def integrate(img, binner):
+    return binner(img)
+
+
+def polarization_correction(img, geo, polarization_factor=.99):
+    return img / geo.polarization(img.shape, polarization_factor)
+
+
+def div(img, count):
+    return img / count
+
+
+db.add_filter(bt_piLast='Billinge')
+hdrs = db(start_time=1496855060.5545955 - 10000)
```

Comment:
`hdrs` is an iterable, which means that I can iterate over it as needed. Note that the code performs a `tee` to as to not remove the data from the object.
https://github.com/NSLS-II/databroker/blob/master/databroker/broker.py#L149